astropy/units/equivalencies.py

```diff
@@ -24,22 +26,71 @@
            'pixel_scale', 'plate_scale', 'with_H0']
 
 
+class Equivalency(UserList):
+    """
+    A container for a units equivalency.
+
+    Attributes
+    ----------
+    name: `str`
+        The name of the equivalency.
+    args: `tuple`
+        Any arguments used to make the equivalency.
+    kwargs: `dict`
+        Any keyword arguments used to make the equivalency.
+    """
+    def __init__(self, equiv_list, name='', args=None, kwargs=None):
+        self.data = equiv_list
+        self.name = [name]
+        self.args = [args] if args is not None else [tuple()]
+        self.kwargs = [kwargs] if kwargs is not None else [dict()]
+
+
+    def __add__(self, other):
+        new = super().__add__(other)
+        new.name = self.name
+        new.args = self.args
+        new.kwargs = self.kwargs
+        if isinstance(other, Equivalency):
+            new.name += other.name
+            new.args += other.args
+            new.kwargs += other.kwargs
+        return new
+
+    def __eq__(self, other):
+        equal = True
+        if not isinstance(other, self.__class__):
+            return False
+        else:
+            if self.name != other.name:
+                equal = False
+            if self.args != other.args:
+                return False
+
+            for i in range(len(self.name)):
```

Comment:
What's wrong with just doing `if self.kwargs != other.kwargs`? I thought the list equality would check the individual dicts for you.